### ISSUE TITLE
Renaming Group to Role?

### ISSUE THREAD

**Issue by marc-farre** [humhub/humhub#6782]
This is an open discussion.

### Is your feature request related to a problem? Please describe.

Most people new to Humhub are confused with the "Group" term.
Many thinks that a group is a space.

Reasons:
- Humhub "Groups" are used to give permissions to users, whereas it is usually roles that assign permissions
- on Facebook, a "Group" is the equivalent of a "Space" on Humhub
- I don't know other plaforms using the term "Group" for roles (e.g. SalesForce, Mastodon, Mattermost, Rocket.Chat, BuddyPress, bitrix24, etc. and even MacOs(1) and Windows) except Nexcloud (but it's not a social platform, so it's not confusing) 
- On Mattermost, groups are used for LDAP sync (maybe it's the reason why Nextcloud uses groups?)

Moreover, Humhub already uses the "Role" term in spaces, which is also to assign permissions.

### Describe the solution you'd like

Renaming Groups with Roles in:
- Translations
- DB tables
- Models
- Controllers
- Views
- Tests
- etc.

Which is quite a big work. But i would be good for Humhub's future, and the more we wait to change it, the more difficult it is, because some modules depends on Groups.

### Describe alternatives you've considered

Renaming only in translations, but I don't think it's a good idea for the code logic.

(1) On MacOS admin also has groups, allowing assigning a role to a group or users.

**Comment by ArchBlood** [comment]
I don't mind the idea, my only concern would be the translations that could also confuse the users;

### Example
#### German Language Translation
`Role => Rolle`
> Rolle: A character or part. It can also refer to a relatively small wheel on which something is rolled, such as on a wheelie bin. In drama and cinematography, it can also mean a roller, castor, or pulley.

**Comment by felixhahnweilheim** [comment]
I am German speaking and I am pretty sure it won't be confused with wheels.
We actually use "Rolle" for Administrator, Moderator etc.
And we have the same problem that Facebook uses the word "Gruppe" (=group) for what we call Space.

Example from Facebook help page:

> EN: There are two **roles** for people who manage groups: admins and moderators.

> DE: Es gibt zwei **Rollen** für Personen, die Gruppen verwalten: Admins und Moderator*innen.

So also in German the renaming would make it easier for new users to intuitively understand the system.

**Comment by luke-** [comment]
The term "role" is definitely better for the Permission perspective.

 But we also use groups for things like:

- Default Spaces
- Registration and Approval Process
- User "Group" Admins
- "Grouping" Members on the "People" page

Not sure the term "Role" would be fit into this usecases.

**Comment by marc-farre** [comment]
@luke- yes, I agree, mainly for default spaces.

However, we still have the confusions mentioned in my first post and I think that these features could be for roles.

About default spaces, an idea would be to move this setting in the Space advanced settings, adding a field with "Role" picker called something like "Roles whose members are to be added to this space".

The main difficulty I think is all the consequences in the code and for existing Humhub instances (maybe some users wouldn't want to change for roles).

I suggest to leave this feature request open to see if we have reactions of other users later.

**Comment by luke-** [comment]
@Semir1212 What do you think about it?

**Comment by Semir1212** [comment]
@luke- I agree, role would fit better. Ideally we would also separate the permissions from the rest, but I think that would annoy to many people.

**Comment by turbostas1** [comment]
Progress always annoys many people because they get used to things, but after some time they will wonder how they at all survived before that change 😀 

For me, changing to Roles would be a massive improvement.

**Comment by martin-rueegg** [comment]
I personally think that "group" is a very generic term. So it should not be used just on its own. Hence, neither should the special meaning in facebook (as being also a space) be applied to such a generic term. Particularly, since actual end users are not really confronted with the therm "group" in humhub anyway, are they? Is it not mostly for administrators? If the latter is the case, we cannot assume that "Most people new to Humhub are confused with the 'Group' term." It would, if so, only apply to admins.

Having said that, I have also been coming across these terms when working with module permissions for spaces. Because there, there are "fixed-named" groups `owner`, `admin`, `member`, `user` etc. These, in fact, seem to be roles of a given user in relation to the given space.

In Linux and Windows, the term _group_ is indeed also used for "user groups". So maybe, the term `group` in Humhub could simply be replaced with the term `User Group` in the UI (and in the code base moving forward). Because that is what they do: group users together.

A "role" is always in relation to another entity. So one person can have the role of CEO in one company, but customer in another. Likewise, a humhub user can be an admin/owner of one space, but not another. As such, these specific pre-defined "groups" could very well be seen as roles in relation to spaces. However, the user groups (currently just "groups") are system-wide. If a user is in the user group `very_special_users` this is true whatever space a member of this group is currently viewing. Hence, a distinction between the user-defined `user groups` and the system-defined `space roles` would really make sense.

I would actually suggest to change those "group names" into integers (with the current contant names) and make them compatible with the current humhub groups. To avoid conflicts with existing groups, negative integers could be used. This would allow to also give specific permissions to any user group in relation to a space (or module in that space) - not only to those fixed "roles".

**Comment by marc-farre** [comment]
@martin-rueegg Thanks for sharing your analysis.

Of course, each instance has a different use case and is configured very differently from one to the other.
So it is always difficult to find terms that fit everyone.

In my experience, many Humhub maintainers want to have a custom term for Spaces and Groups, which is not possible on Humhub, or can be done more or less by overwriting translations.
So in an ideal world, these terms should be configurable.

What I see from the instances I know is that groups are mainly created to give different permissions to users. We very often have `System Administrators`, `Administrators`, `Moderators` and `Users` or `Members`. So "Role" is better here.
But sometimes it's used to classify the community in the same way as its organisation (e.g. geographical regions, internal departments, or internal hierarchy). In this case, your suggested "User Group" is a better fit.

So maybe "User Group" is a better idea. It's also a solution to avoid confusion with Spaces. I've changed this issue title to add this suggestion.

In fact, Spaces already uses User Group in the code: https://github.com/humhub/humhub/blob/a75592b65b94afe58855bc1414995db107d93b89/protected/humhub/modules/space/models/Space.php#L81-L86

As for the idea of changing the name of Space roles to integer, I'm not sure we should spend time on that, because:
- I personally prefer names, as it's easier for a human to read the database directly.
- Names as constants are already widely used in Humhub.
- Maybe one day we'll want to refactor Space Roles to be able to add custom roles, which means creating a new table for them and maybe removing these constants.
